homeassistant/components/oralb/__init__.py

```diff
@@ -25,14 +29,55 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     address = entry.unique_id
     assert address is not None
     data = OralBBluetoothDeviceData()
+
+    def _needs_poll(
+        service_info: BluetoothServiceInfoBleak, last_poll: float | None
+    ) -> bool:
+        # Only poll if hass is running, we need to poll,
+        # and we actually have a way to connect to the device
+        return (
+            hass.state == CoreState.running
+            and data.poll_needed(service_info, last_poll)
+            and bool(
+                async_ble_device_from_address(
+                    hass, service_info.device.address, connectable=True
+                )
+            )
+        )
+
+    async def _async_poll(service_info: BluetoothServiceInfoBleak):
+        # BluetoothServiceInfoBleak is defined in HA, otherwise would just pass it
+        # directly to the Xiaomi code
```

Comment:
Xiaomi?

Reply:
Its a stale copy.  It should have been `oralb` code.

@conway220 Would you please fix in a fresh PR?